scripts/west_commands/runners/spi_burn.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021, Telink Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import time
+import subprocess
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+class SpiBurnBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for SPI_burn.'''
+
+    def __init__(self, cfg, addr, eraseall, spiburn, iceman):
+        super().__init__(cfg)
+
+        self.bin = cfg.bin_file
+        self.spiburn = spiburn
+        self.iceman = iceman
+        self.addr = addr
+        if eraseall:
+            self.eraseall = "--erase-all"
+        else:
+            self.eraseall = ''
+
+    @classmethod
+    def name(cls):
+        return 'spi_burn'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--addr', default='0x0',
+                            help='start flash address to write')
+        parser.add_argument('--erase-all', default=False, action='store_true',
+                            help='erase flash before burn')
+        parser.add_argument('--telink-tools-path', help='path to Telink flash tools')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+
+        if args.telink_tools_path:
+            spiburn = f'{args.telink_tools_path}/flash/bin/SPI_burn'
+            iceman = f'{args.telink_tools_path}/ice/ICEman'
+        else:
+            # If telink_tools_path arg is not specified then pass to tools shall be specified in PATH
+            spiburn = 'SPI_burn'
+
+            # Find path to ICEman tool
+            process = subprocess.Popen(['which', 'ICEman'],
```

Comment:
Thanks for the update. I have no objections to allowing an override of the tools path, but please just use `self.require` to find executables on the system. This helper function returns the path to the executable it finds, and takes care of the error handling if it's not found.

```py
iceman = self.require('ICEman')
```

Reply:
@mbolivar-nordic Got it! Thanks for the tip. Not it is implemented with usage of **required**.